File: c/gpt/p78.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    int status;
    char name[50];
} Person;

int main(void) {
    /* Allocate memory for a Person (malloc returns a pointer) */
    Person *p = malloc(sizeof *p);
    if (!p) {
        perror("malloc");
        return 1;
    }

    /* Use -> to access members through the pointer */
    p->status = 0;
    strcpy(p->name, "John");

    printf("Person: status=%d, name=%s\n", p->status, p->name);

    free(p);
    return 0;
}

File: java/gpt/p78.java
public class p78 {
    static class Person {
        int status;
        String name;
    }

    public static void main(String[] args) {
        // Allocate a new Person (Java's `new` is similar to memory allocation)
        Person p = new Person();
        p.status = 0;
        p.name = "John";

        System.out.printf("Person: status=%d, name=%s%n", p.status, p.name);
    }
}
